### DISCUSSION TITLE
How can I add a truncated version of Positive prompt to filename?

### DISCUSSION THREAD

**Discussion by tusharbhutt** [WASasquatch/was-node-suite-comfyui#146] [Q&A] [answered]
AS per subject, I'd like to add the positive prompt (limited by X words or Y character length) to the filename when output.  I'm thinking I have to add a custom text node?   How can it pull the prompt needed, and more importantly, can it be limited in length?

Thanks!

TB

**Comment by WASasquatch** [comment]
The widgets meta inputs don't seem to be working right in custom nodes so a theoretical node would just have to accept a text input that you also input into a CLIPTextEncode or whatever. Adding a truncations node shouldn't be that hard. Just have a mode for character/word and number for truncations and swap functionality by mode.

**Reply by tusharbhutt** [reply]
Thanks, I will futz around in the various nodes and see what jumps out, I don't have the skills in python (or any language for that matter) to program a node myself.  I think I can look for a node that has some kind of length limitation  and try to strip out the underlying code :)

**Comment by WASasquatch** [comment, marked as answer]
I've added a node for this: https://github.com/WASasquatch/was-node-suite-comfyui/commit/fb53a8aed2826de5155f99c47a1bafd3c62d7d26

![Screenshot 2023-07-30 210518](https://github.com/WASasquatch/was-node-suite-comfyui/assets/1151589/b5054623-1760-4241-bb0f-ae77b2f1e310)
![Screenshot 2023-07-30 210547](https://github.com/WASasquatch/was-node-suite-comfyui/assets/1151589/48befa74-0168-4005-851d-53341e072b66)

**Reply by tusharbhutt** [reply]
Wow, thanks for doing that, I was not expecting something at all.  I found this [link](https://blenderneko.github.io/ComfyUI-docs/Interface/SaveFileFormatting/#search-and-replace-strings) for adding node/widget names to filenames so hopefully that will do the trick

**Reply by WASasquatch** [reply]
> I found this [link](https://blenderneko.github.io/ComfyUI-docs/Interface/SaveFileFormatting/#search-and-replace-strings) for adding node/widget names to filenames so hopefully that will do the trick

Apparently it has some issues working in custom nodes from what I've been reported.

**Reply by tusharbhutt** [reply]
Yep,  it did not work with a primitive node (probably because they don't have S&R names, at least that I can see).  I'll have to test on other nodes later

**Reply by tusharbhutt** [reply]
FYI, this workfow works! Then you put the token name in as part of the filename.  I **did** have trouble when I changed the token but I hd also removed the filename delimiter character and it royally messed up the naming convention, but I don't know which of those issues is the root, so I am leaving it alone for now

![image](https://github.com/WASasquatch/was-node-suite-comfyui/assets/26264462/a24252c9-52dc-45e6-96db-38d64c81aacd)
